lib/create-contentful-api.js

```diff
@@ -1,73 +1,73 @@
+export class ClientAPI {
+  /**
+   * @param {Object} params - API initialization params
+   * @param {Object} params.http - HTTP client instance
+   */
+  constructor ({ http }) {
+    this._http = http
+  }
 
   /**
    * Gets all spaces
-   * @memberof ClientAPI   * @return {Promise<Space.SpaceCollection>} Promise for a collection of Spaces
+   * @return {Promise<SpaceCollection>} Promise for a collection of Spaces
    * @example
    * client.getSpaces()
    * .then(spaces => console.log(spaces.items))
    */
-  function getSpaces () {
-    return http.get('')
-    .then((response) => wrapSpaceCollection(http, response.data), errorHandler)
+  getSpaces () {
+    console.dir(this._http)
+    return this._http.get('')
+    .then((response) => wrapSpaceCollection(this._http, response.data), errorHandler)
   }
 
   /**
    * Gets a space
-   * @memberof ClientAPI
    * @param {string} id - Space ID
-   * @return {Promise<Space.Space>} Promise for a Space
+   * @return {Promise<Space, ErrorResponse>} Promise for a Space
    * @example
    * client.getSpace('spaceid')
    * .then(space => console.log(space))
    */
-  function getSpace (id) {
-    return http.get(id)
-    .then((response) => wrapSpace(http, response.data), errorHandler)
+  getSpace (id) {
+    return this._http.get(id)
+    .then((response) => wrapSpace(this._http, response.data), errorHandler)
   }
 
   /**
    * Creates a space
-   * @memberof ClientAPI
-   * @see {Space.Space}
+   * @see {Space}
    * @param {object} data - Object representation of the Space to be created
-   * @param {string=} organizationId - Organization ID, if the associated token can manage more than one organization.
-   * @return {Promise<Space.Space>} Promise for the newly created Space
+   * @param {string} [organizationId] - Organization ID, if the associated token can manage more than one organization.
+   * @return {Promise<Space>} Promise for the newly created Space
    * @example
    * client.createSpace({name: 'Space Name'})
    * .then(space => console.log(space))
    */
-  function createSpace (data, organizationId) {
-    return http.post('', data, {
+  createSpace (data, organizationId) {
+    return this._http.post('', data, {
       headers: organizationId ? {'X-Contentful-Organization': organizationId} : {}
     })
-    .then((response) => wrapSpace(http, response.data), errorHandler)
+    .then((response) => wrapSpace(this._http, response.data), errorHandler)
   }
+}
 
-  return {
-    getSpaces: getSpaces,
-    getSpace: getSpace,
-    createSpace: createSpace
-  }
+/**
+ * Creates API object with methods to access functionality from Contentful's
+ * Management API
+ *
+ * @param {Object} params - API initialization params
+ * @param {Object} params.http - HTTP client instance
+ * @param {Function} params.shouldLinksResolve - Link resolver preconfigured with global setting
+ * @return {ClientAPI}
+ */
+export default function createClientAPI (params) {
```

Comment:
`createClientAPI` should be `@private` so we should not expose in the documentation, as it will be confusing with `createClient`